cli/cmd/cli_state.go

```diff
@@ -430,6 +430,8 @@ func (c *cliState) envs() []string {
 		fmt.Sprintf("LW_NOCOLOR=%s", os.Getenv("NO_COLOR")),
 		fmt.Sprintf("LW_LOG=%s", c.LogLevel),
 		fmt.Sprintf("LW_JSON=%v", c.jsonOutput),
+		fmt.Sprintf("LW_PROFILE=%s", c.Profile),
```

Comment:
I don't think we should pass `LW_PROFILE` 

I think the CLI has the logic to parse the profile and pass the necessary information to
the component, the problem I have is for example, if we pass the profile foo via `--profile`
but also change the sub-account via `--subaccount`, how does the component know
when to use the profile and when to use the rest of the flags?

Do you think we should still pass it?

Reply:
To answer your narrow question the component knows because it looks at LW_SUBACCONT.

The IAC component has its own configuration which it stores in its own config file -- the most important config value is the IAC organization to use.  This will be the wrong IAC org if you run `lacework configure switch-profile`.

The IAC component can store which profile it was used with, and if it's different then it can do something other than proceed blindly ahead with what surely will be wrong values.

For the most part this are issues that only internal users (or those with multiple profiles) will hit.  But it'll be really confusing when they do hit them.

Reply:
I don't know much about the iac component, but FWIW it would seem that `LW_PROFILE` is a CLI implementation detail and that `LW_ACCOUNT` and `LW_SUBACCOUNT` are what's important here...